services/worker-manager/src/providers/aws.js

```diff
@@ -254,6 +254,34 @@ class AwsProvider extends Provider {
     }));
   }
 
+  async removeWorker({worker}) {
+    let result;
+    try {
+      result = await this.ec2s[worker.providerData.region].terminateInstances({
+        InstanceIds: [worker.workerId]
+      }).promise();
+    } catch (e) {
+      this.monitor.err(`Error terminating AWS instance: ${e}`);
```

Comment:
The call to `workerPool.reportError` will log the error as a `workerError` message, so I think this is redundant.

Reply:
We also don't want to necessarily report an error at all. If the instance is already gone, this should just be a noop. So I think a 404 or whatever is fine.